### ISSUE TITLE
The nnet3-copy with prepare_for_test=true  doesn't collapse the model as old version.

### ISSUE THREAD

**Issue by JiayiFu** [kaldi-asr/kaldi#3661]
I trained a CNN-TDNNF model and used nnet3-copy --binary=false prepare_for_test=true to 
collapse it.
I compared the results of prepare_for_test=true and prepare_for_test=false and I found it just change the test mode: 'F' -> 'T'. It doesn't modify the network structure like the old version.
In the old version, there is a line in the text model mdl:
`component-node name=tdnnf7.linear component=cnn6.batchnorm.tdnnf7.linear input=cnn6.relu`
which combines two components together to speed up.
But now, the result is still two lines in the file:
`component-node name=cnn6.batchnorm component=cnn6.batchnorm input=cnn6.relu
component-node name=tdnnf7.linear component=tdnnf7.linear input=cnn6.batchnorm`

Why it doesn't modify the structure now?

**Comment by danpovey** [comment]
After some recent changes to the nnet computation-graph code (#3531), I discovered that the model-collapsing code was causing a bug.  I believe it was causing wrong output for a TDNN-F model.  It seemed to have something to do with a sequence of a dropout layer and then a batchnorm layer, or vice versa.  I found that if I turned off either the batchnorm collapsing or the dropout collapsing, the behavior disappeared.  As a result, I turned off both of those collapse modes by default in CollapseModelConfig(), see #3565.  My intention was to reverse that change after finding the bug, but I was not able to find the bug and have not had the time to revisit it.
It should be safe to turn one of those two back on.

**Comment by jtrmal** [comment]
Is this ok to close?

**Comment by JiayiFu** [comment]
> Is this ok to close?
This issue(prepare_for_test=true doesn't collapse the model) doesn't affect the normal use except that it may slow down the inference speed a little. 
I think is ok to close.

**Comment by danpovey** [comment]
I think I'd rather leave it open.  It is actually a bug that is bothering me, I simply haven't had time to do it (and with next-gen Kaldi stuff on the table, I may never have time to do it.)
